### PR TITLE
PLANNER-2583: Add docs for Benchmarker properties

### PR DESCRIPTION
**Pull request by Christopher-Chianelli**
Note: Spring reuses the Solver Termination Config while Quarkus does not, meaning Spring technically has `best-score-limit` and `unimproved-time-spent` properties for benchmarker whereas Quarkus does not.

### JIRA

https://issues.redhat.com/browse/PLANNER-2583

### Checklist
- [ ] Documentation updated if applicable.
- [ ] Upgrade recipe provided if applicable.

<details>
<summary>
How to replicate CI configuration locally?
</summary>

We do "simple" maven builds, they are just basically maven commands, but just because we have multiple repositories related between them and one change could affect several of those projects by multiple pull requests, we use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.

[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is not only a github-action tool but a CLI one, so in case you posted multiple pull requests related with this change you can easily reproduce the same build by executing it locally. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>specific pull request build</b> please add comment: <b>Jenkins (re)run [optaplanner|kogito-apps|kogito-examples|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] tests</b>
* for a <b>full downstream build</b> 
  * for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  * for <b>github actions</b> job: add the label `run_fdb`
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
* for a <b>Quarkus LTS check</b> please add comment: <b>Jenkins run LTS</b>
* for a <b>specific Quarkus LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|kogito-apps|kogito-examples|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] LTS</b>
* for a <b>Native check</b> please add comment: <b>Jenkins run native</b>
* for a <b>specific Native LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|kogito-apps|kogito-examples|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] native</b>
</details>

### CODE REVIEW

File: optaplanner-docs/src/modules/ROOT/pages/integration/config-properties.adoc [hunk 1]
```diff
@@ -58,3 +58,17 @@ Terminates the solver when a specific or higher score has been reached.
 For example: `0hard/-1000soft` terminates when the best score changes from `0hard/-1200soft` to `0hard/-900soft`.
 Wildcards are supported to replace numbers.
 For example: `0hard/*soft` to terminate when any feasible score is reached.
+
+{property_prefix}optaplanner.benchmark.solver-benchmark-config-xml::
+A classpath resource to read the solver configuration XML.
```
Comment (ge0ffrey) [inline optaplanner-docs/src/modules/ROOT/pages/integration/config-properties.adoc:63]: Is that a solver configuration? I thought it was a benchmark configuration.

Comment (ge0ffrey) [reply optaplanner-docs/src/modules/ROOT/pages/integration/config-properties.adoc:63]: Are we sure we want to write this property name in stone? Clearly, it's ambiguous?

Comment (ge0ffrey) [reply optaplanner-docs/src/modules/ROOT/pages/integration/config-properties.adoc:63]: Ok, the javadoc API in Quarkus OptaPlannerBenchmarkBuildTimeConfig also has this problem.
And the Spring BenchmarkProperties, which I added it in, has it too.

But it's not a solve configuration, in any of these cases. The naming is confusing.

Comment (ge0ffrey) [reply optaplanner-docs/src/modules/ROOT/pages/integration/config-properties.adoc:63]: I'll set up a meeting on how to handle this...

Comment (ge0ffrey) [reply optaplanner-docs/src/modules/ROOT/pages/integration/config-properties.adoc:63]: Meeting decided that we don't need to go further than current state.
@Christopher-Chianelli Would you mind changing this docs line and the javadoc lines for quarkus/spring to from "solver configuration XML" to:
```suggestion
A classpath resource to read the benchmark configuration XML.
```

**Review by ge0ffrey — APPROVED** [review]
Good to merge once javadocs and docs for benchmark configuration xml don't say its a solver configuration xml

File: optaplanner-docs/src/modules/ROOT/pages/integration/config-properties.adoc (hunk 1 above)
Comment (Christopher-Chianelli) [reply optaplanner-docs/src/modules/ROOT/pages/integration/config-properties.adoc]: Done; Spring was already benchmark

**Pull request merged**